src/cdc_uart.c

```diff
@@ -66,80 +74,102 @@ void cdc_uart_init(void) {
     gpio_set_dir(PICOPROBE_UART_DTR, GPIO_OUT);
     gpio_put(PICOPROBE_UART_DTR, 1);
 #endif
+#if (CDC_UARTS > 1)
+    gpio_set_function(PICOPROBE_EX_UART_TX, GPIO_FUNC_UART);
+    gpio_set_function(PICOPROBE_EX_UART_RX, GPIO_FUNC_UART);
+    gpio_set_pulls(PICOPROBE_EX_UART_TX, 1, 0);
+    gpio_set_pulls(PICOPROBE_EX_UART_RX, 1, 0);
+    uart_init(PICOPROBE_EX_UART_INTERFACE, PICOPROBE_EX_UART_BAUDRATE);
+#endif
 }
 
-void cdc_task(void)
+void cdc_task(uint8_t tty)
 {
-    static int was_connected = 0;
-    static uint cdc_tx_oe = 0;
     uint rx_len = 0;
 
+    uart_inst_t* uart_ptr = PICOPROBE_UART_INTERFACE;
+#if (CDC_UARTS > 1)
+    if (tty != 0) { uart_ptr = PICOPROBE_EX_UART_INTERFACE; }
+#endif
+
     // Consume uart fifo regardless even if not connected
-    while(uart_is_readable(PICOPROBE_UART_INTERFACE) && (rx_len < sizeof(rx_buf))) {
-        rx_buf[rx_len++] = uart_getc(PICOPROBE_UART_INTERFACE);
+    while(uart_is_readable(uart_ptr) && (rx_len < RX_BUF_SIZ)) {
+        rx_buf[tty][rx_len++] = uart_getc(uart_ptr);
     }
 
-    if (tud_cdc_connected()) {
-        was_connected = 1;
+    if (tud_cdc_n_connected(tty)) {
+        was_connected[tty] = 1;
         int written = 0;
         /* Implicit overflow if we don't write all the bytes to the host.
          * Also throw away bytes if we can't write... */
         if (rx_len) {
+          if (!tty) {
 #ifdef PICOPROBE_UART_RX_LED
-          gpio_put(PICOPROBE_UART_RX_LED, 1);
-          rx_led_debounce = debounce_ticks;
+            gpio_put(PICOPROBE_UART_RX_LED, 1);
+            rx_led_debounce = debounce_ticks;
 #endif
-          written = MIN(tud_cdc_write_available(), rx_len);
+          }
+          written = MIN(tud_cdc_n_write_available(tty), rx_len);
           if (rx_len > written)
-              cdc_tx_oe++;
+              cdc_tx_oe[tty]++;
 
           if (written > 0) {
-            tud_cdc_write(rx_buf, written);
-            tud_cdc_write_flush();
+            tud_cdc_n_write(tty, rx_buf[tty], written);
+            tud_cdc_n_write_flush(tty);
           }
         } else {
+          if (!tty) {
 #ifdef PICOPROBE_UART_RX_LED
-          if (rx_led_debounce)
-            rx_led_debounce--;
-          else
-            gpio_put(PICOPROBE_UART_RX_LED, 0);
+            if (rx_led_debounce)
+              rx_led_debounce--;
+            else
+              gpio_put(PICOPROBE_UART_RX_LED, 0);
 #endif
+          }
         }
 
       /* Reading from a firehose and writing to a FIFO. */
-      size_t watermark = MIN(tud_cdc_available(), sizeof(tx_buf));
+      size_t watermark = MIN(tud_cdc_n_available(tty), TX_BUF_SIZ);
       if (watermark > 0) {
         size_t tx_len;
+        if (!tty) {
 #ifdef PICOPROBE_UART_TX_LED
-        gpio_put(PICOPROBE_UART_TX_LED, 1);
-        tx_led_debounce = debounce_ticks;
+          gpio_put(PICOPROBE_UART_TX_LED, 1);
```

Comment:
I wonder if it _might_ be useful to have e.g. `PICOPROBE_EX_UART_TX_LED` (etc.) which defaults to `PICOPROBE_UART_TX_LED` if not defined? :thinking:

Reply:
I wouldn't use it, but happy to add - again let me know your preference

Reply:
actually, those are `debugprobe`-only, which conflicts with one of the default UART0 pins (and would be tricky to solder the extra 2 uart tx/rx wires to), so I think its better if I make the `picoprobe`-only nature of `EXTRA_UART` explicit in `picoprobe_config.h` - does this sound OK?

Reply:
I'll also change all instances of `CDC_UARTS > 1` to `CDC_UARTS == 2`
which makes more sense / is less confusing once the compiler range check you suggested is in place